### PR TITLE
Replace jaxb with DOM in PolicyConstraint{Value}

### PR DESCRIPTION
**Pull request by ckelleyRH**

### CODE REVIEW

File: base/common/src/main/java/com/netscape/certsrv/profile/PolicyConstraintValue.java [hunk 1]
```diff
@@ -120,18 +121,119 @@ public boolean equals(Object obj) {
...
+
     public String toXML() throws Exception {
-        Marshaller marshaller = JAXBContext.newInstance(PolicyConstraintValue.class).createMarshaller();
-        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+        Document document = builder.newDocument();
+
+        Element accountElement = toDOM(document);
+        document.appendChild(accountElement);
```
Comment (edewata) [inline base/common/src/main/java/com/netscape/certsrv/profile/PolicyConstraintValue.java:215]: We probably should call it `rootElement` or just `element`.

File: base/common/src/main/java/com/netscape/certsrv/profile/PolicyConstraintValue.java [hunk 2]
```diff
@@ -120,18 +121,119 @@ public boolean equals(Object obj) {
...
 
     public static PolicyConstraintValue fromXML(String xml) throws Exception {
-        Unmarshaller unmarshaller = JAXBContext.newInstance(PolicyConstraintValue.class).createUnmarshaller();
-        return (PolicyConstraintValue) unmarshaller.unmarshal(new StringReader(xml));
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+        Document document = builder.parse(new InputSource(new StringReader(xml)));
+
+        Element accountElement = document.getDocumentElement();
+        return fromDOM(accountElement);
```
Comment (edewata) [inline base/common/src/main/java/com/netscape/certsrv/profile/PolicyConstraintValue.java:236]: Same thing here.

**Review by edewata — APPROVED** [review]
Just a nit, but it LGTM.

**Comment by ckelleyRH** [comment]
Hmm, let's not get too excited... failed the QE sanity check :thinking:

**Comment by edewata** [comment]
I restarted the QE tests to make sure that it's not an intermittent problem. The IPA tests passed so that's a good sign.

There's a possibility we'd have to change some other classes at the same time. Or maybe remove JAXB annotations only after the JAXB -> DOM conversion is complete.

**Comment by edewata** [comment]
Yeah, restoring JAXB annotations seems to fix it. I'm not sure why, but I think we can merge the other changes first to narrow down the possibilities.

**Comment by ckelleyRH** [comment]
It's probably because I need to do `PolicyConstraint` at the same time.

**Comment by ckelleyRH** [comment]
Still failing, I think I may have to also do `ProfileData` to make this one pass.

**Comment by ckelleyRH** [comment]
I think the issue is a request problem, like @edewata was mentioning earlier on anotherPR:
`172.17.0.2 - caadmin [19/Aug/2021:11:04:02 +0000] "POST /ca/rest/agent/certrequests/26/approve HTTP/1.1" 400 161`
which when we look through could be `CertReviewResponse` not playing ball.

File: base/common/src/main/java/com/netscape/certsrv/profile/PolicyConstraintValue.java (hunk 1 above)
Comment (ckelleyRH) [reply base/common/src/main/java/com/netscape/certsrv/profile/PolicyConstraintValue.java:215]: Renamed

Comment (ckelleyRH) [reply base/common/src/main/java/com/netscape/certsrv/profile/PolicyConstraintValue.java:215]: As above

**Comment by ckelleyRH** [comment]
I've reinstated the jaxb annotations so this should pass the CI now.

**Review by c-dorney — APPROVED** [review]

**Comment by ckelleyRH** [comment]
Thanks!

**Pull request merged**